File: RealEstate.Domain/Properties/Property.cs
using RealEstate.Domain.Entity;

namespace RealEstate.Domain.Properties
{
    public class Property : BaseEntity
    {
        public string? Title { get;  set; }
        public string? Description { get;  set; }
        public string? Location { get;  set; }
        public string? Image {  get;  set; }
        public decimal Price { get;  set; }
        public PropertyStatus Status { get;  set; }
        public int RealtorId { get;  set; }
    }

    public enum PropertyStatus
    {
        Available,
        Reserved,
        Sold
    }

}

File: RealEstate.Presentation/RealEstate/src/app/models/property/property.ts
export enum PropertyStatus {
    Available = 'Available',
    Reserved = 'Reserved',
    Sold = 'Sold',
  }
  
export class Property {
    id: string;
    title?: string;
    description?: string;
    location?: string;
    image?: string;
    price: number;
    status: PropertyStatus;
    realtorId: number;
  
    constructor(
      id: string,
      price: number,
      status: PropertyStatus,
      realtorId: number,
      title?: string,
      description?: string,
      location?: string,
      image?: string
    ) {
      this.id = id;
      this.title = title;
      this.description = description;
      this.location = location;
      this.image = image;
      this.price = price;
      this.status = status;
      this.realtorId = realtorId;
    }
  }
  